pkg/proxy/proxy.go

```diff
@@ -91,25 +94,38 @@ func (s *Server) Run() error {
 		return err
 	}
 
-	return http.Serve(l, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if r.Method != http.MethodConnect {
-			http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
-			return
-		}
-
-		ip, _, err := net.SplitHostPort(r.Host)
-		if err != nil {
-			http.Error(w, err.Error(), http.StatusBadRequest)
-			return
-		}
-
-		if !subnet.Contains(net.ParseIP(ip)) {
-			http.Error(w, http.StatusText(http.StatusForbidden), http.StatusForbidden)
-			return
-		}
-
-		Proxy(s.Log, w, r, 0)
-	}))
+	return http.Serve(l, http.HandlerFunc(s.proxyHandler))
+}
+
+func (s Server) proxyHandler(w http.ResponseWriter, r *http.Request) {
+	err := s.validateProxyResquest(w, r)
+	if err != nil {
+		return
+	}
+	Proxy(s.Log, w, r, 0)
+}
+
+// validateProxyResquest checks that the request is valid. If not, it writes the
+// appropriate http headers and returns an error.
+func (s Server) validateProxyResquest(w http.ResponseWriter, r *http.Request) error {
+
+	ip, _, err := net.SplitHostPort(r.Host)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return errors.New("Request is not valid")
```

Comment:
Good catch , I got a little to eager with the copy/pasting. I made the error messages more precise on the other two also. 

Maybe we should also log the error in the proxy handler too. I didn't include it at first because I did not want to modify the behaviour, but it seems that it makes sense.

Reply:
Good catch , I got a little too eager with the copy/pasting. I made the error messages more precise on the other two also.